pkg/trace/api/api.go

```diff
@@ -383,30 +387,62 @@ func (r *HTTPReceiver) tagStats(v Version, header http.Header) *info.TagStats {
 	})
 }
 
-func decodeTraces(v Version, req *http.Request) (*decodedTraces, error) {
+// decodeTracerPayload decodes the payload in http request `req`.
+// - tp is the decoded payload
+// - ranHook reports whether the decoder was able to run the pb.MetaHook
+// - err is the first error encountered
+func decodeTracerPayload(v Version, req *http.Request, ts *info.TagStats) (tp *pb.TracerPayload, ranHook bool, err error) {
 	switch v {
 	case v01:
 		var spans []pb.Span
-		if err := json.NewDecoder(req.Body).Decode(&spans); err != nil {
-			return nil, err
+		if err = json.NewDecoder(req.Body).Decode(&spans); err != nil {
+			return nil, false, err
 		}
-		return &decodedTraces{
-			Traces: tracesFromSpans(spans),
-		}, nil
+		return &pb.TracerPayload{
+			LanguageName:    ts.Lang,
+			LanguageVersion: ts.LangVersion,
+			ContainerID:     req.Header.Get(headerContainerID),
+			Chunks:          traceChunksFromSpans(spans),
+			TracerVersion:   ts.TracerVersion,
+		}, false, nil
 	case v05:
 		buf := getBuffer()
 		defer putBuffer(buf)
-		if _, err := io.Copy(buf, req.Body); err != nil {
-			return nil, err
+		if _, err = io.Copy(buf, req.Body); err != nil {
+			return nil, false, err
 		}
 		var traces pb.Traces
-		err := traces.UnmarshalMsgDictionary(buf.Bytes())
-		return &decodedTraces{
-			Traces:  traces,
-			RanHook: true,
-		}, err
+		err = traces.UnmarshalMsgDictionary(buf.Bytes())
+		return &pb.TracerPayload{
+			LanguageName:    ts.Lang,
+			LanguageVersion: ts.LangVersion,
+			ContainerID:     req.Header.Get(headerContainerID),
+			Chunks:          traceChunksFromTraces(traces),
+			TracerVersion:   ts.TracerVersion,
+		}, true, err
+	case v06:
+		buf := getBuffer()
+		defer putBuffer(buf)
+		if _, err = io.Copy(buf, req.Body); err != nil {
+			return nil, false, err
+		}
+		var tracerPayload pb.TracerPayload
+		err = tracerPayload.Unmarshal(buf.Bytes())
+		return &tracerPayload, true, err
```

Comment:
Since this returns `true` for `ranHook`. I'd like us to 100% ensure this and not risk any issues in this area. As you can imagine that would be quite unfortunate. Do you mind adding an extra sub-test to `pkg/trace/test/testsuite/cards_test.go` for this? You can use this function for the new endpoint: https://github.com/DataDog/datadog-agent/blob/552a135e24478923e35266057531105a1ab6e28e/pkg/trace/test/runner.go#L102-L106 It should work fine.